server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java

```diff
@@ -610,32 +590,37 @@ private CandidateOpportunity updateCandidateOpportunity(
 
             final CandidateOpportunityStage stage = oppParams.getStage();
             if (stage != null) {
-                // If stage changing, automate post to JobCreatorSourcePartner chat
+                // Stage is changing
                 if (!oppParams.getStage().equals(opp.getStage())) {
-
-                    // Find the relevant job chat
-                    JobChat jcspChat = jobChatService.getOrCreateJobChat(
-                        JobChatType.JobCreatorSourcePartner,
-                        opp.getJobOpp(),
-                        opp.getCandidate().getUser().getPartner(),
-                        null
-                    );
-
-                    // Set the chat post content
-                    Post autoPostCandidateOppStageChange = new Post();
-                    autoPostCandidateOppStageChange.setContent(
-                        "💼 <b>" + opp.getName() + "</b> 🪜<br> This case for candidate "
-                            + candidateNameAndNumber
-                            + " has changed stage from '" + opp.getStage() + "' to '"
-                            + oppParams.getStage() + "'."
-                    );
-
-                    // Create the chat post
-                    ChatPost candidateOppStageChangeChatPost = chatPostService.createPost(
-                        autoPostCandidateOppStageChange, jcspChat, userService.getSystemAdminUser());
-
-                    // Publish the chat post
-                    chatPostService.publishChatPost(candidateOppStageChangeChatPost);
+                    // If stage is changing to CLOSED (e.g. removed from submission list) automate removal post.
+                    if (stage.isClosed()) {
+                        publishRemovedFromSubmissionListPosts(opp, stage);
+                    } else {
+                        // If non closing stage change, automate post to JobCreatorSourcePartner chat
+                        // Find the relevant job chat
```

Comment:
Given that you have other nice publish methods,  maybe it would be good to put this non closing stage change  post into its own method. It might make the code look simpler.

Reply:
I held back as this was Sam's code, but now I see you like the private publish methods I feel ok to put this into it's own private method.

Reply:
@camerojo made the changes as requested